crates/viewer/re_space_view_graph/src/ui/draw.rs

```diff
@@ -192,24 +192,43 @@ fn draw_arrow(painter: &Painter, tip: Pos2, direction: Vec2, color: Color32) {
 }
 
 /// Draws an edge between two points, optionally with an arrow at the target point.
-fn draw_edge(ui: &mut Ui, points: [Pos2; 2], show_arrow: bool) -> Response {
+pub fn draw_edge(ui: &mut Ui, geometry: &EdgeGeometry, show_arrow: bool) -> Response {
     let fg = ui.style().visuals.text_color();
+    let stroke = Stroke::new(1.0, fg);
 
-    let rect = Rect::from_points(&points);
     let painter = ui.painter();
-    painter.line_segment(points, Stroke::new(1.0, fg));
 
-    // Calculate direction vector from source to target
-    let direction = (points[1] - points[0]).normalized();
+    match geometry.path {
+        PathGeometry::Line { source, target } => {
+            painter.line_segment([source, target], stroke);
+        }
+        PathGeometry::CubicBezier {
+            source,
+            target,
+            control,
+        } => {
+            painter.add(CubicBezierShape {
+                points: [source, control[0], control[1], target],
+                closed: false,
+                fill: Color32::TRANSPARENT,
+                stroke: stroke.into(),
+            });
+        }
+    }
 
     // Conditionally draw an arrow at the target point
     if show_arrow {
-        draw_arrow(painter, points[1], direction, fg);
+        draw_arrow(
+            painter,
+            geometry.target_pos(),
+            geometry.target_arrow_direction(),
+            stroke.color,
+        );
     }
 
     // We can add interactions in the future, for now we simply allocate the
     // rect, so that bounding boxes are computed correctly.
-    ui.allocate_rect(rect, NON_SENSE)
+    ui.allocate_rect(geometry.bounding_rect(), NON_SENSE)
```

Comment:
idiomatically, `NON_SENSE` is `egui::Sense::hover()`. Also, nonsense sounds... weird :)

Reply:
Yeah that was a pun that somehow made it through the previous review process, I've removed the code 😇.

Reply:
This is where kurbo can help. It can convert various things to cubic bezier, and can also linearise everything to polylines.